src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -1632,4 +1645,217 @@ private static unsafe IntPtr ObjectToComWeakRef(object target, out long wrapperI
+                            m_value = entry.m_value,
+                            m_next = newBuckets[bucket],
+                        };
+                        newBuckets[bucket] = newEntry;
+
+                        entry = nextEntry;
+                    }
+                }
+                _buckets = newBuckets;
+            }
+            catch (OutOfMemoryException)
+            {
+            }
+        }
+
+        public void Remove(GCHandle handle)
+        {
+            using (_lock.EnterScope())
+            {
+                int bucket = GetBucket(handle);
+                Entry? prev = null;
+                Entry? entry = _buckets[bucket];
+                while (entry != null)
+                {
+                    if (handle.Equals(entry.m_value))
+                    {
+                        if (prev == null)
+                        {
+                            _buckets[bucket] = entry.m_next;
+                        }
+                        else
+                        {
+                            prev.m_next = entry.m_next;
+                        }
+                        _numEntries--;
+                        return;
+                    }
+
+                    prev = entry;
+                    entry = entry.m_next;
+                }
+            }
+        }
+
+        private int GetBucket(GCHandle handle, int numBuckets = 0)
+        {
+            int h = handle.GetHashCode();
+            h &= 0x7fffffff;
+            return (h % (numBuckets == 0 ? _buckets.Length : numBuckets));
+        }
+
+        public Enumerator GetEnumerator() => new Enumerator(this);
+
+        IEnumerator<GCHandle> IEnumerable<GCHandle>.GetEnumerator() => GetEnumerator();
+
+        IEnumerator IEnumerable.GetEnumerator() => ((IEnumerable<GCHandle>)this).GetEnumerator();
+
+        private sealed class Entry
+        {
+            public GCHandle m_value;
+            public Entry? m_next;
+        }
+
+        public struct Enumerator : IEnumerator<GCHandle>
+        {
+            private readonly Entry?[] _buckets;
+            private int _currentIdx;
+            private Entry? _currentEntry;
+
+            public Enumerator(GCHandleSet set)
+            {
+                // We hold onto the buckets of the set rather than the set itself
+                // so that if it is ever expanded, we are not impacted by that during
+                // enumeration.
+                _buckets = set._buckets;
+                Reset();
+            }
+
+            public GCHandle Current
+            {
+                get
+                {
+                    if (_currentEntry == null)
+                    {
+                        throw new InvalidOperationException("InvalidOperation_EnumOpCantHappen");
+                    }
+
+                    return _currentEntry.m_value;
+                }
+            }
+
+            object IEnumerator.Current => Current;
+
+            public void Dispose()
```

Comment:
Is this needed?

Reply:
It is needed given we implement IEnumerable<T>.

Reply:
`IEnumerable` doesn't have a `Dispose()` method nor does it require implementing `IDisposable`. I'm not sure why this is required.

Reply:
IEnumerator (which this also implements) requires IDisposable.

Reply:
Ah. Thanks.